lib/src/models/moderator.dart

```diff
@@ -0,0 +1,51 @@
+import 'dart:convert';
+
+class Moderator {
+  final String name;
+  final String authorFlairText;
+  final List<String> modPermissions;
+  final DateTime date;
+  final String relId;
+  final String id;
+  final String authorFlairCssClass;
+
+  Moderator(
+    this.name,
+    this.authorFlairText,
+    this.modPermissions,
+    this.date,
+    this.relId,
+    this.id,
+    this.authorFlairCssClass,
+  );
+
+  Map<String, dynamic> toMap() {
+    return {
+      'name': name,
+      'authorFlairText': authorFlairText,
+      'modPermissions': modPermissions,
+      'date': date?.millisecondsSinceEpoch,
+      'relId': relId,
+      'id': id,
+      'authorFlairCssClass': authorFlairCssClass,
+    };
+  }
+
+  static Moderator fromMap(Map<String, dynamic> map) {
```

Comment:
Let's make this a named constructor instead of a static method. We can assert that `map != null` in the initialization list and eventually we'll be able to remove this check once Dart moves to a null safe type system in the near future.

Also, for future reference, I typically avoid the pattern `if (x == null) return null` when I can unless `null` is a valid return. It's more informative to throw [ArgumentError.notNull](https://api.dart.dev/stable/2.8.2/dart-core/ArgumentError/ArgumentError.notNull.html) instead.